core/templates/dev/head/editor/feedback_tab/FeedbackTab.js

```diff
@@ -121,20 +127,31 @@ oppia.controller('FeedbackTab', [
         $scope.cancelReview = function() {
           $modalInstance.dismiss('cancel');
         };
+
+        $scope.suggestionValid = function() {
+          return (activeThread.status === 'open' &&
+            explorationData.data.version === suggestion.exploration_version);
+        }
       }]
     }).result.then(function(result) {
       threadDataService.resolveSuggestion(
-        $scope.activeThread.suggestion.suggestion.id, result.action);
-      // Update the status of the feedback thread.
-      var msg, status;
-      if (result.action === 'accept') {
-        msg = 'Suggestion accepted.';
-        status = 'fixed';
-      } else {
-        msg = 'Suggestion rejected.';
-        status = 'not_actionable';
-      }
-      $scope.addNewMessage($scope.activeThread.thread_id, msg, status);
+        $scope.activeThread.thread_id, result.action, result.commitMsg,
+        function(res) {
+          threadDataService.fetchThreads(function() {
+            $scope.setActiveThread($scope.activeThread.thread_id);
+          });
+          // Immediately update editor to reflect accepted suggestion.
+          if(result.action === 'accept') {
+            var suggestion = $scope.activeThread.suggestion;
+            var stateName = suggestion.state_name;
+            var state = angular.copy(explorationData.data.states[stateName]);
+            state.content[0].value = suggestion.state_content.value;
+            explorationStatesService.setState(stateName, state);
+            $rootScope.$broadcast('refreshStateEditor');
```

Comment:
I've implemented a check to prevent acting on suggestions when there are unsaved changes: see be0fd91